### PR TITLE
[Bug] [core] Cannot load plugin due to plugin name is not class name

### LINKED ISSUE #1662: [Bug] [core] Cannot load plugin
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.

### What happened

This bug is introduced by #1592. 
Right now, we will use the plugin class name to judge the plugin which should be used.
This will cause problem, since our plugin name is different with the class name.

### SeaTunnel Version

dev

### SeaTunnel Config

```conf
no
```

### Running Command

```shell
no
```

### Error Exception

```log
ClassNotFoundException("Plugin class not found by name :["xx"]")
```

### Flink or Spark Version

_No response_

### Java or Scala Version

_No response_

### Screenshots

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by ruanwenjun**
## Purpose of this pull request
close #1662 
This pr add a new method in Plugin.
```java
   /**
     * Return the plugin name, this is used in seatunnel conf DSL.
     *
     * @return plugin name.
     */
    default String getPluginName() {
        return this.getClass().getSimpleName();
    }
```
And will use this method to charge whether the plugin should be used. And override this method in subclass to adapte the current plugin name.

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in you PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/development/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

**Review by Hisoka-X — DISMISSED** [review]
LGTM

File: seatunnel-connectors/seatunnel-connectors-spark/seatunnel-connector-spark-jdbc/src/main/scala/org/apache/seatunnel/spark/jdbc/sink/Jdbc.scala [hunk 1]
```diff
@@ -69,4 +69,6 @@ class Jdbc extends SparkBatchSink {
         "duplicateIncs" -> ""))
     config = config.withFallback(defaultConfig)
   }
+
+  override def getPluginName: String = "jdbc"
```
Comment (Hisoka-X) [inline seatunnel-connectors/seatunnel-connectors-spark/seatunnel-connector-spark-jdbc/src/main/scala/org/apache/seatunnel/spark/jdbc/sink/Jdbc.scala]: Maybe all plugin name use upper camel case would be better on code specification？

Comment (ruanwenjun) [reply seatunnel-connectors/seatunnel-connectors-spark/seatunnel-connector-spark-jdbc/src/main/scala/org/apache/seatunnel/spark/jdbc/sink/Jdbc.scala]: Agree with you, done.

**Comment by ruanwenjun** [comment]
This is strange, I can build success in my laptop with the same command.

**Review by CalvinKirs — COMMENTED** [review]
Maybe we need to update plugin docs?

**Comment by ruanwenjun** [comment]
> Maybe we need to update plugin docs?

No need to update the docs, since the  plugin name is still the old.

**Review by CalvinKirs — APPROVED** [review]

**Pull request merged**